src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ProjectRazorJsonPublisher.cs

```diff
@@ -286,15 +286,15 @@ protected virtual void SerializeToFile(ProjectSnapshot projectSnapshot, string p
             {
                 var projectRazorJson = new ProjectRazorJson(publishFilePath, projectSnapshot);
                 _serializer.Serialize(writer, projectRazorJson);
+            }
 
-                var fileInfo = new FileInfo(publishFilePath);
-                if (fileInfo.Exists)
-                {
-                    fileInfo.Delete();
-                }
+            var fileInfo = new FileInfo(publishFilePath);
```

Comment:
This is not necessarily part of the fix, but it seemed like closing the stream earlier was a good thing.